### PR TITLE
Prevent /tmp from filling up.

### PR DESCRIPTION
**Pull request by MorganEPatch**
For some reason, every time chef runs the `npm -g install npm` bash block in this cookbook, a folder containing the unpacked tarball of the newest npm version is left in /tmp. Npm _should_ be cleaning this folder up to prevent high disk usage, but for some reason it doesn't when run from chef. This is a temporary fix to prevent Chef from filling up the /tmp folder by deleting these folders manually.

### CODE REVIEW

File: providers/default.rb [hunk 1]
```diff
@@ -54,6 +54,13 @@
     code 'npm -g install npm'
   end
 
+  # hacky fix to prevent filling up /tmp due to bug with npm
+  execute 'Clear temp files' do
+    command "find . -depth -iname 'npm-*' -type d -exec rm -r \"{}\" \\;"
```
Comment (ramereth) [inline providers/default.rb]: I don't feel good about this. Can you set it directly to `find /tmp ...`. I know that cwd is set to it, but it'll make me feel better if the command actually does what you're wanting.

**Review by ramereth — CHANGES_REQUESTED** [review]
Tests?

File: providers/default.rb (hunk 1 above)
Comment (MorganEPatch) [reply providers/default.rb]: Fixed that and added tests.

**Comment by ramereth** [comment]
@LyonesGamer can you also add unit tests for this change please?

File: test/integration/default/serverspec/test_tmp_clear_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'serverspec'
+set :backend, :exec
+
+# Test that the git repository's directory is properly set up
+describe command('ls /tmp | grep "npm-"') do
+  its(:stdout) { should match '' }
+  its(:exit_status) { should eq 0 }
```
Comment (ramereth) [inline test/integration/default/serverspec/test_tmp_clear_spec.rb]: You actually want this to return 1. See the grep man page for more details.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by ramereth — APPROVED** [review]

**Comment by LyonesGamer** [comment]
!bump minor

**Pull request merged**